src/entity/base_entity.rs

```diff
@@ -745,10 +745,77 @@ pub trait EntityTrait: EntityName {
     fn delete_many() -> DeleteMany<Self> {
         Delete::many(Self::default())
     }
+
+    /// Delete a model based on primary key
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use sea_orm::{error::*, tests_cfg::*, *};
+    /// #
+    /// # #[smol_potat::main]
+    /// # #[cfg(feature = "mock")]
+    /// # pub async fn main() -> Result<(), DbErr> {
+    /// #
+    /// # let db = MockDatabase::new(DbBackend::Postgres)
+    /// #     .append_exec_results(vec![
+    /// #         MockExecResult {
+    /// #             last_insert_id: 0,
+    /// #             rows_affected: 1,
+    /// #         },
+    /// #     ])
+    /// #     .into_connection();
+    /// #
+    /// use sea_orm::{entity::*, query::*, tests_cfg::fruit};
+    ///
+    /// let delete_result = fruit::Entity::delete_by_id(1).exec(&db).await?;
+    ///
+    /// assert_eq!(delete_result.rows_affected, 1);
+    ///
+    /// assert_eq!(
+    ///     db.into_transaction_log(),
+    ///     vec![Transaction::from_sql_and_values(
+    ///         DbBackend::Postgres,
+    ///         r#"DELETE FROM "fruit" WHERE "fruit"."id" = $1"#,
+    ///         vec![1i32.into()]
+    ///     )]
+    /// );
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn delete_by_id(values: <Self::PrimaryKey as PrimaryKeyTrait>::ValueType) -> DeleteMany<Self> {
+        let mut delete = Self::delete_many();
+        let mut keys = Self::PrimaryKey::iter();
+        for v in values.into_value_tuple() {
+            if let Some(key) = keys.next() {
+                let col = key.into_column();
+                delete = delete.filter(col.eq(v));
+            } else {
+                panic!("primary key arity mismatch");
+            }
+        }
+        if keys.next().is_some() {
+            panic!("primary key arity mismatch");
+        }
+        delete
+    }
 }
 
 #[cfg(test)]
 mod tests {
+    #[test]
+    fn test_delete_by_id() {
+        use crate::tests_cfg::cake;
+        use crate::{entity::*, query::*, DbBackend};
+        assert_eq!(
+            cake::Entity::delete_by_id(1)
+                .build(DbBackend::Sqlite)
+                .to_string(),
+            r#"DELETE FROM "cake" WHERE "cake"."id" = 1"#,
+        );
+    }
+    
```

Comment:
Is there a reason as to why `"cake"."id"` is written separately in quotation marks?

Reply:
I used [this](https://github.com/SeaQL/sea-orm/blob/0a26dee674a00cdda4cf3568f3d89a76fb5c8f57/src/query/delete.rs#L182-L219) as a reference.

Reply:
Quotation marks were added around identifier, `"cake"` is a identifier as it's a table name and `"id"` is a identifier as well as it's a column name.
See https://www.sqlite.org/lang_keywords.html for more

Reply:
I am aware about the identifier patterns, I don't think its a common pattern with Postgres/MySQL thats why I wasn't expecting this